File: flight-copilot-ui/src/types/flight.ts
export type AirportOption = {
  iata: string;
  icao?: string;
  name: string;
  city?: string;
  country?: string;
  label?: string; // server returns label if you added it
};

export type FlightSearchPayload = {
  origin: string; // IATA
  destination: string; // IATA
  departureDate: string; // YYYY-MM-DD
  returnDate?: string; // YYYY-MM-DD
  maxPrice?: number;
  nonStop?: boolean;
  currency?: string;
  max?: number;
};

export type ResponseFlightSearch = {
  options: IFlight[];
  output?: string;
};

// The  interfaces of the response back from server
export interface IFlightPrice {
  amount: number;
  currency: string;
}

export interface ISegmentView {
  origin: string;
  destination: string;
  depart_utc: string;
  arrive_utc: string;
  carrier: string;
  flight_number?: string;
  duration_min: number;
}

export interface ILayoverView {
  at: string;
  duration_min: number;
}

export interface ILeg {
  origin: string;
  destination: string;
  depart_utc: string;
  arrive_utc: string;
  duration_min: number;
  stops: number;
  segments: ISegmentView[];
  layovers: ILayoverView[];
}

export interface IFlight {
  price: IFlightPrice;
  deeplink: null;
  carriers: string[];
  outbound: ILeg;
  return_: ILeg;
}

File: src/schemas/flight.py
from pydantic import BaseModel, Field
from typing import Optional, List


class FlightRequest(BaseModel):
    origin: str = Field(...,
                        description="IATA code or city text (TLV or 'Tel Aviv')")
    destination: str = Field(...,
                             description="IATA code or city text (PRG or 'Prague')")
    departureDate: str = Field(...,
                               description="YYYY-MM-DD or month word like 'april'")
    returnDate: Optional[str] = Field(
        None, description="YYYY-MM-DD or month word")
    maxPrice: Optional[int] = Field(None, ge=1)
    nonStop: Optional[bool] = False
    currency: Optional[str] = "USD"
    max: Optional[int] = 10


class AgentRequest(BaseModel):
    query: str


class Price(BaseModel):
    amount: int
    currency: str


class SegmentView(BaseModel):
    origin: str            # IATA
    destination: str       # IATA
    depart_utc: str        # ISO string
    arrive_utc: str        # ISO string
    carrier: str
    flight_number: Optional[str] = None
    duration_min: int


class LayoverView(BaseModel):
    at: str                # IATA
    duration_min: int


class Leg(BaseModel):
    origin: str            # leg start IATA
    destination: str       # leg end IATA
    depart_utc: str
    arrive_utc: str
    duration_min: int
    stops: int
    segments: List[SegmentView]
    layovers: List[LayoverView]


class FlightOption(BaseModel):
    price: Price
    deeplink: Optional[str] = None
    carriers: List[str]
    outbound: Optional[Leg] = None
    return_: Optional[Leg] = None

    class Config:
        populate_by_name = True


class FlightResponse(BaseModel):
    options: List[FlightOption]


class AgentResponse(BaseModel):
    options: List[FlightOption]
    output: Optional[str]
